### PR TITLE
[Vidio] Add new extractor (Closes #7195)

### LINKED ISSUE #7195: Feature Request: Support for vidio.com site
Hi all,

I want to download video from vidio.com use youtube-dl but never success.
Hope anyone can add the vidio.com to youtube-dl feature

Thanks

### PR DESCRIPTION
**Pull request by TRox1972**

### CODE REVIEW

File: youtube_dl/extractor/vidio.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        formats = self._extract_m3u8_formats(
+            video_data['sources'][0]['file'],
+            display_id, ext='mp4')
+
+        return {
+            'id': video_id,
+            'title': self._og_search_title(webpage),
+            'formats': formats,
+            'thumbnail': video_data.get('image', ''),
```
Comment (yan12125) [inline youtube_dl/extractor/vidio.py]: Incorrect fallback value.

File: youtube_dl/extractor/vidio.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id, display_id = mobj.group('id', 'display_id')
+
+        webpage = self._download_webpage(url, display_id)
+
+        video_data = self._parse_json(self._html_search_regex(
+            r'data-json-clips\s*=\s*"\[(.+)\]"', webpage, display_id), display_id)
```
Comment (yan12125) [inline youtube_dl/extractor/vidio.py]: The third parameter of ```_html_search_regex``` is name but not ID.

File: youtube_dl/extractor/vidio.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+from __future__ import unicode_literals
+
+import re
+from .common import InfoExtractor
+
+from ..utils import int_or_none
+
+
+class VidioIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?vidio\.com/watch/(?P<id>\d{6})-(?P<display_id>[^/?]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/vidio.py]: That's incorrect. `id` may be arbitrary length https://www.vidio.com/watch/77949-south-korea-test-fires-missile-that-can-strike-all-of-the-north.

File: youtube_dl/extractor/vidio.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    _TEST = {
+        'url': 'http://www.vidio.com/watch/165683-dj_ambred-booyah-live-2015',
+        'info_dict': {
+            'id': '165683',
+            'title': 'DJ_AMBRED - Booyah (Live 2015)',
+            'ext': 'mp4',
+            'thumbnail': 'https://cdn0-a.production.vidio.static6.com/uploads/video/image/165683/dj_ambred-booyah-live-2015-bfb2ba.jpg',
```
Comment (dstftw) [inline youtube_dl/extractor/vidio.py]: Use more relaxed regex.

**Comment by dstftw** [comment]
Cherry-picked in 7def35712a7047578643f18eaf6dda79fd8c9291, thanks.

**Comment by TRox1972** [comment]
@dstftw Thanks for accepting the PR. Could you close issue #7195 as the request is now fulfilled?

**Pull request closed without merging**